nixos/tests/kernel-generic.nix

```diff
@@ -7,6 +7,89 @@
 with pkgs.lib;
 
 let
+  patchedPkgs = pkgs.extend (
+    final: prev: {
+      linuxKernel = prev.linuxKernel // {
```

Comment:
Surprising to me that this isn't a .override interface.  Usually when you do a merge of something overridable like this, it means recursive updates won't take place so other stuff won't be overridden properly.